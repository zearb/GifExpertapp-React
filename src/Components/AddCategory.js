import React,{useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        // console.log(inputValue);
        if(inputValue.trim().length>1){
            
            setCategories( categ => [inputValue,...categ] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p style={{display:'none'}}> {inputValue} </p>
            <input type="text" value={inputValue} onChange={handleInputChange} />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
} 