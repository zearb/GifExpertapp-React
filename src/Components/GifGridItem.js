import React from 'react';
import PropTypes from 'prop-types';

export default function GifGridItem({title,url}) {

    
    // console.log(props);

    return (
        <div className="card animate__animated animate__zoomIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}