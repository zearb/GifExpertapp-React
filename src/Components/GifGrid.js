import React,{} from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data:images,loading} = useFetchGifs(category);
    // console.log(data);


    return (
        <div>
            <h3 className="animate__animated animate__zoomIn">{category}</h3>

            {loading && <p className="animate__animated animate__flash animate__infinite infinite animate__slow	2s">Cargando...</p>}

            <div className="cont_cards">
                {
                    images.map( img => 
                    
                        <GifGridItem
                            key = {img.id}
                            {...img}
                        />

                    )
                }
            </div>
        </div>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}