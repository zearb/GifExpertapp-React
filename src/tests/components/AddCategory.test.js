import React from 'react';
// import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { AddCategory } from '../../Components/AddCategory';

describe('Pruebas en AddCategory', () => {
    
    const setCategories = jest.fn();
    let wrapper;

    beforeEach(()=>{

        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />)

    });
    
    test('AddCategory debe mostrarse correctamente', ()=> {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe Cambiar la caja de texto',()=>{

        const input = wrapper.find('input');
        const value = 'Hola mundo';

        input.simulate('change',{target: {value} });

        expect(wrapper.find('p').text().trim()).toBe(value);

    }); 

    test('No debe de procesar la información al hacer submit',() => {

        wrapper.find('form').simulate('submit',{preventDefault(){}});

        expect(setCategories).not.toHaveBeenCalled();

    });

    test('Debe llamar a setCategories y limpiar la caja de texto', () => {

        const input = wrapper.find('input');
        input.simulate('change',{target: {value:'Hola mundo'} });

        wrapper.find('form').simulate('submit',{preventDefault(){}});

        expect(setCategories).toHaveBeenCalled(); //evalua que se haya llamado la funcion
        expect(setCategories).toHaveBeenCalledTimes(1); //evalua que se haya llamado la funcion 1 vez
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); //evalua que se haya llamado la funcion con parámetro de tipo funcion

        expect(input.prop('value')).toBe('');

    });

});
