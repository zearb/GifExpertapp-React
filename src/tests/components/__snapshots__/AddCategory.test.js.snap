// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory AddCategory debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p
    style={
      Object {
        "display": "none",
      }
    }
  >
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
